perun-rpc/src/main/perl/Perun/beans/User.pm

```diff
@@ -181,8 +181,8 @@ sub setIsServiceUser
 	my $self = shift;
 	my $value = shift;
 	$value eq 'true' || $value eq 1 || ref $value eq "JSON::PP::Boolean"
```

Comment:
Sorry I didn't realize previously, but this will not work. Since it sets 'true' when passed value is boolean (but we don't know if original value is true or false, since we check value type (reference) and not value).
You should check the reference first, if boolean, then set passed value to object immediately. Then 'elsif' branch check if value is 'true' or 1, set boolean true to object, otherwise set false. Otherwise always set false.

Same goes for setIsSponsoredUser() method.